lib/src/proto_op_store.rs

```diff
@@ -127,50 +126,48 @@ fn timestamp_from_proto(proto: &crate::protos::op_store::Timestamp) -> Timestamp
 fn operation_metadata_to_proto(
     metadata: &OperationMetadata,
 ) -> crate::protos::op_store::OperationMetadata {
-    let mut proto = crate::protos::op_store::OperationMetadata::new();
-    proto.start_time = MessageField::some(timestamp_to_proto(&metadata.start_time));
-    proto.end_time = MessageField::some(timestamp_to_proto(&metadata.end_time));
-    proto.description = metadata.description.clone();
-    proto.hostname = metadata.hostname.clone();
-    proto.username = metadata.username.clone();
-    proto.tags = metadata.tags.clone();
-    proto
+    crate::protos::op_store::OperationMetadata {
+        start_time: Some(timestamp_to_proto(&metadata.start_time)),
+        end_time: Some(timestamp_to_proto(&metadata.end_time)),
+        description: metadata.description.clone(),
+        hostname: metadata.hostname.clone(),
+        username: metadata.username.clone(),
+        tags: metadata.tags.clone(),
+    }
 }
 
 fn operation_metadata_from_proto(
-    proto: &crate::protos::op_store::OperationMetadata,
+    proto: crate::protos::op_store::OperationMetadata,
 ) -> OperationMetadata {
-    let start_time = timestamp_from_proto(&proto.start_time);
-    let end_time = timestamp_from_proto(&proto.end_time);
-    let description = proto.description.to_owned();
-    let hostname = proto.hostname.to_owned();
-    let username = proto.username.to_owned();
-    let tags = proto.tags.clone();
+    let start_time = timestamp_from_proto(&proto.start_time.unwrap_or_default());
+    let end_time = timestamp_from_proto(&proto.end_time.unwrap_or_default());
     OperationMetadata {
         start_time,
         end_time,
-        description,
-        hostname,
-        username,
-        tags,
+        description: proto.description,
+        hostname: proto.hostname,
+        username: proto.username,
+        tags: proto.tags,
     }
 }
 
 fn operation_to_proto(operation: &Operation) -> crate::protos::op_store::Operation {
-    let mut proto = crate::protos::op_store::Operation::new();
-    proto.view_id = operation.view_id.as_bytes().to_vec();
+    let mut proto = crate::protos::op_store::Operation {
+        view_id: operation.view_id.as_bytes().to_vec(),
+        metadata: Some(operation_metadata_to_proto(&operation.metadata)),
+        ..Default::default()
+    };
     for parent in &operation.parents {
         proto.parents.push(parent.to_bytes());
     }
-    proto.metadata = MessageField::some(operation_metadata_to_proto(&operation.metadata));
     proto
 }
 
-fn operation_from_proto(proto: &crate::protos::op_store::Operation) -> Operation {
+fn operation_from_proto(proto: crate::protos::op_store::Operation) -> Operation {
     let operation_id_from_proto = |parent: &Vec<u8>| OperationId::new(parent.clone());
     let parents = proto.parents.iter().map(operation_id_from_proto).collect();
     let view_id = ViewId::new(proto.view_id.clone());
```

Comment:
Cant the cloning here be removed now that we receive `proto` by value? This comment seems to apply to many `*_from_proto()` functions in this PR.

Reply:
Oops, thought clippy would catch this. Cleaned up a bunch of these.